src/Uno.UI/UI/Xaml/DependencyProperty.cs

```diff
@@ -70,6 +71,13 @@ private DependencyProperty(string name, Type propertyType, Type ownerType, Prope
 
 			// Improve the performance of the hash code by
 			CachedHashCode = _name.GetHashCode() ^ ownerType.GetHashCode();
+
+			DependentProperties = dependentProperties.ToImmutableArray();
```

Comment:
Why not make it an immutable array in the first place and avoid allocating both a list an array?